services/discourse_translator/yandex.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module DiscourseTranslator
+  class Yandex < Base
+    TRANSLATE_URI = "https://translate.yandex.net/api/v1.5/tr.json/translate"
+    DETECT_URI = "https://translate.yandex.net/api/v1.5/tr.json/detect"
+
+    SUPPORTED_LANG = {
+      pt_BR: 'pt',
+      pl_PL: 'pl',
+      no_NO: 'no',
+      fa_IR: 'fa',
+      zh_CN: 'zh',
+      zh_TW: 'zh',
+      tr_TR: 'tr',
+      en_US: 'en',
+      en_GB: 'en',
+      az: 'az',
+      ml: 'ml',
+      sq: 'sq',
+      mt: 'mt',
+      am: 'am',
+      mk: 'mk',
+      en: 'en',
+      mi: 'mi',
+      ar: 'ar',
+      mr: 'mr',
+      hy: 'hy',
+      mhr: 'mhr',
+      af: 'af',
+      mn: 'mn',
+      eu: 'eu',
+      de: 'de',
+      ba: 'ba',
+      ne: 'ne',
+      be: 'be',
+      no: 'no',
+      bn: 'bn',
+      pa: 'pa',
+      my: 'my',
+      pap: 'pap',
+      bg: 'bg',
+      fa: 'fa',
+      bs: 'bs',
+      pl: 'pl',
+      cy: 'cy',
+      pt: 'pt',
+      hu: 'hu',
+      ro: 'ro',
+      vi: 'vi',
+      ru: 'ru',
+      ht: 'ht',
+      ceb: 'ceb',
+      gl: 'gl',
+      sr: 'sr',
+      nl: 'nl',
+      si: 'si',
+      mrj: 'mrj',
+      sk: 'sk',
+      el: 'el',
+      sl: 'sl',
+      ka: 'ka',
+      sw: 'sw',
+      gu: 'gu',
+      su: 'su',
+      da: 'da',
+      tg: 'tg',
+      he: 'he',
+      th: 'th',
+      yi: 'yi',
+      tl: 'tl',
+      id: 'id',
+      ta: 'ta',
+      ga: 'ga',
+      tt: 'tt',
+      it: 'it',
+      te: 'te',
+      is: 'is',
+      tr: 'tr',
+      es: 'es',
+      udm: 'udm',
+      kk: 'kk',
+      uz: 'uz',
+      kn: 'kn',
+      uk: 'uk',
+      ca: 'ca',
+      ur: 'ur',
+      ky: 'ky',
+      fi: 'fi',
+      zh: 'zh',
+      fr: 'fr',
+      ko: 'ko',
+      hi: 'hi',
+      xh: 'xh',
+      hr: 'hr',
+      km: 'km',
+      cs: 'cs',
+      lo: 'lo',
+      sv: 'sv',
+      la: 'la',
+      gd: 'gd',
+      lv: 'lv',
+      et: 'et',
+      lt: 'lt',
+      eo: 'eo',
+      lb: 'lb',
+      jv: 'jv',
+      mg: 'mg',
+      ja: 'ja',
+      ms: 'ms',
+    }
+
+    def self.access_token_key
+      "yandex-translator"
+    end
+
+    def self.access_token
+      SiteSetting.translator_yandex_api_key || (raise TranslatorError.new("NotFound: Yandex API Key not set."))
+    end
```

Comment:
`access_token` is expected to be called by specs, but isn't used anywhere. Shouldn't it be called in `default_query` (instead of `SiteSetting.translator_yandex_api_key`)?

Reply:
Right, my bad, should be fixed now